src/arch/amdgpu/common/tlb.hh

```diff
@@ -55,336 +55,353 @@
 
 namespace gem5
 {
-
 class BaseTLB;
 class Packet;
 class ThreadContext;
 
 namespace X86ISA
 {
-    class GpuTLB : public ClockedObject
-    {
-      protected:
-        friend class Walker;
-
-        typedef std::list<TlbEntry*> EntryList;
+class GpuTLB : public ClockedObject
+{
+  protected:
+    friend class Walker;
 
-        uint32_t configAddress;
+    typedef std::list<TlbEntry *> EntryList;
 
-      public:
-        typedef X86GPUTLBParams Params;
-        GpuTLB(const Params &p);
-        ~GpuTLB();
+    uint32_t configAddress;
 
-        typedef enum BaseMMU::Mode Mode;
+  public:
+    typedef X86GPUTLBParams Params;
+    GpuTLB(const Params &p);
+    ~GpuTLB();
 
-        class Translation
-        {
-          public:
-            virtual ~Translation() { }
-
-            /**
-             * Signal that the translation has been delayed due to a hw page
-             * table walk.
-             */
-            virtual void markDelayed() = 0;
-
-            /**
-             * The memory for this object may be dynamically allocated, and it
-             * may be responsible for cleaning itslef up which will happen in
-             * this function. Once it's called the object is no longer valid.
-             */
-            virtual void finish(Fault fault, const RequestPtr &req,
-                                ThreadContext *tc, Mode mode) = 0;
-        };
-
-        void dumpAll();
-        TlbEntry *lookup(Addr va, bool update_lru=true);
-        void setConfigAddress(uint32_t addr);
-
-      protected:
-        EntryList::iterator lookupIt(Addr va, bool update_lru=true);
-        Walker *walker;
+    typedef enum BaseMMU::Mode Mode;
 
+    class Translation
+    {
       public:
-        Walker *getWalker();
-        void invalidateAll();
-        void invalidateNonGlobal();
-        void demapPage(Addr va, uint64_t asn);
-
-      protected:
-        int size;
-        int assoc;
-        int numSets;
-
-        /**
-         *  true if this is a fully-associative TLB
-         */
-        bool FA;
-        Addr setMask;
+        virtual ~Translation() {}
 
         /**
-         * Allocation Policy: true if we always allocate on a hit, false
-         * otherwise. Default is true.
+         * Signal that the translation has been delayed due to a hw page
+         * table walk.
          */
-        bool allocationPolicy;
+        virtual void markDelayed() = 0;
 
         /**
-         * if true, then this is not the last level TLB
+         * The memory for this object may be dynamically allocated, and it
+         * may be responsible for cleaning itslef up which will happen in
+         * this function. Once it's called the object is no longer valid.
          */
-        bool hasMemSidePort;
+        virtual void finish(Fault fault, const RequestPtr &req,
```

Comment:
Is aligning the args to the left parenthesis the rule that led to the least amount of immediate changes in the codebase?

I ask this because since the alignment depends on the length of the function´s name/return type, it becomes variable, possibly requiring re-alignment in the future in case the function´s name/return type changes. I am not a fan of this approach, and I prefer only adding one level of indentation to the continuation lines, but this is just my opinion, so you can disregard it :)

Reply:
> Is aligning the args to the left parenthesis the rule that led to the least amount of immediate changes in the codebase?
> 

Yes as most of the code was written in this way

> I ask this because since the alignment depends on the length of the function´s name/return type, it becomes variable, possibly requiring re-alignment in the future in case the function´s name/return type changes. I am not a fan of this approach, and I prefer only adding one level of indentation to the continuation lines, but this is just my opinion, so you can disregard it :)

Yes I agree that aligning is not the best for the reason you mentioned (on top of which I would also add merge/rebase conflicts). However at the moment changing the option would make the diff bigger so I would keep it as it is if you don't mind

Reply:
If the plan is to keep it as is and not change to the new-line rule, then that is the sensible approach; however, if the plan is to later create a PR to change to use the new-line rule, then that means that we will generate merge conflicts on the same spots twice, which does not seem to be a good idea.